client/tx/query.go

```diff
@@ -48,50 +49,44 @@ func QueryTxCmd(cdc *codec.Codec) *cobra.Command {
 	return cmd
 }
 
-func queryTx(cdc *codec.Codec, cliCtx context.CLIContext, hashHexStr string) ([]byte, error) {
+func queryTx(cdc *codec.Codec, cliCtx context.CLIContext, hashHexStr string) (out sdk.TxResponse, err error) {
 	hash, err := hex.DecodeString(hashHexStr)
 	if err != nil {
-		return nil, err
+		return out, err
 	}
 
 	node, err := cliCtx.GetNode()
 	if err != nil {
-		return nil, err
+		return out, err
 	}
 
 	res, err := node.Tx(hash, !cliCtx.TrustNode)
 	if err != nil {
-		return nil, err
+		return out, err
 	}
 
-	if !cliCtx.TrustNode {
-		err := ValidateTxResult(cliCtx, res)
-		if err != nil {
-			return nil, err
-		}
+	if err = ValidateTxResult(cliCtx, res); !cliCtx.TrustNode && err != nil {
```

Comment:
This does not equal to what was before. Before we didn't execute `ValidateTxResult` if `TrustNode` == true.

Reply:
But we were checking if `!clictx.TrustNode` to enter this loop so this is the same right? Am I missing something?

Reply:
good catch.